### ISSUE TITLE
網頁代碼撥放器

### ISSUE THREAD

**Issue by LilyLiuweb** [LilyLiuweb/youtube#3]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css"> 
    
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
    
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <title>Learning jQuery UI</title>
   
    <style>
        body{
            margin: 0;
            padding: 0;
            background-color: #1D1E22;
            overflow: hidden;
        }
        #menu{

            width: 100%;
            height: 60px;
            background-color: black;
            font-family:"Helvetica Neue","Helvetica","Arial";
        }
        .menu-container{

            width: 1200px;
            margin: auto;
            padding-top: 15px;
        }
        #logo{
            color:white;
            float:left;
            font-size: 1.5em;
            margin-left: 500px;
            padding-left: 15px;
        }

        .button{

            float: right;
            font-size: 18px;
            width: 120px;
            border:auto;
            padding: 5px;
            border-radius: 5px;
            background-color:#FFDD40;

            cursor: pointer;
            transition: .5s;
            

        }
        .button span{
            transition: .5s;

        }

        .button span::after {
            content: "\00bb";
            /* color: white; */
            position: absolute;
            opacity: 0;
            right:0px;
            transition: .5s;
        }
        .button:hover span{

            position:relative;
            padding-right: 25px;

        }
        .button:hover span::after{

            opacity: 1;
            right: 0;

        }
        .clear{

            clear:both;

        }
        #accordion{
            
            background-color:#FFDD40;
            width:100%;
            height:400px;
            margin: auto;
            overflow: hidden;
        }
        #accordion h3, #accordion div{
         /* 為了讓h3和textarea放在同一行的關鍵設定 */
            display:inline-block;
        /* 為了讓textarea靠容器最上方填滿 */
            vertical-align: top;
            text-align: center ;
        }
        #accordion h3{
            background-color: #36383F;
            color:white;
        /* 為了讓標題從橫的變為垂直顯示 */
            transform:rotate(270deg);

            width: 400px;
            height: 30px;

            margin: 185px 0px 0px -170px;

        }
        #accordion div{

            width: 300px;
            height: 400px;
            margin: 0px 5px 0px -170px;
        }
        textarea{
            width:100%;
            /* 繼承inherit父級高度 */
            height : inherit; 
            /* 不讓用戶自訂輸入框大小 */
            resize : none;
            background-color: inherit;
            border: none;
            color:#ddca7e;
            font-size: 20px;
        }
        #result-container{
            width:100%; 
            background-color:#1D1E22;             
        }
        h2{
            margin:0;
            width:100%;
            height:30px;

            text-align: center; 

            color:white;
            background-color: #36383F;
           
            font-family:"Helvetica Neue","Helvetica","Arial";
            font-weight: normal;
        }
        iframe{
            width:100%;
            border:none;
            background-color: white;
        }
        footer{
            background-color: #36383F;
            width:100%;
            height:40px;
            position: fixed;
        }
    </style>
</head>

<body>

    <div id="menu">
        <div class="menu-container">

            <div id="logo">網頁代碼撥放器</div>

            <button id="runCode" type="button" class="button" name="button"><span>運行代碼</span></button>
        </div>

    </div>

    <div class="clear"></div>

    <ul id="accordion">
        <h3 id="html-code" class="tab">HTML</h3>
        <div class="code-container">
            <textarea name="" id="html-code" cols="30" rows="10">

            </textarea>
        </div>
        <h3 id="css-code" class="tab">CSS</h3>
        <div class="code-container">
            <textarea name="" id="css-code" cols="30" rows="10">

            </textarea>
        </div>
        <h3 id="js-code" class="tab">JavaScript</h3>
        <div class="code-container">
            <textarea name="" id="js-code" cols="30" rows="10">

            </textarea>
        </div>

    </ul>

    <dic class="clear"></dic>

    <div id="result-container">
        <h2 class="result">結 果</h2>
        <iframe id="result-frame">

        </iframe>
    </div>

    <div id="footer"></div>

    <script type="text/javascript">
        //將容器預設寬度去掉
        $(".code-container").hide();
        //改變css跟js的位置html維持
        $("#css-code").css("margin-left","-350px");
        $("#js-code").css("margin-left","-350px");
        // 收起所有textarea的動畫
        $(".tab").click(function(event){

            // 所指定元素之下一個所有的元素 nextAll
            $(this).nextAll(".tab").css("margin-left","-170px");
            
            $(this).nextAll(".code-container").animate(
            //    放入兩個參數: {屬性1, "數值1"},{運行時間, queue:repeat}
                {width:"0px"},
                {duration: 1500 , queue: false}

            );

            $(this).next().animate(
                {
                    backgroundColor: "#1D1E22",
                }
            );
                //offset 方法 可返回或設置 指定元素 偏移(位置)
                //返回的對象包含 兩個屬性 : top 和 left ,以像素計
              var leftOffset = $(this).offset().left;
            
            if(leftOffset > 680){
                // $(this).prevAll(".tab").css("margin-left","-170px");

                $(this).prevAll(".tab").animate(
                    {
                        //不是margin-left
                        marginLeft:"-170px",
                    },
                    {
                        duration: 700 , queue: false
                    }
                );

                $(this).next().animate(
                {
                    width:"680px"
                },
                {
                    duration: 2000 , queue: false
                }
            );
            // prevAll : .prevAll( [selector ] )
            //          Get all preceding siblings of each element 
            //          in the set of matched elements, optionally filtered by a selector.
                $(this).prevAll(".code-container").animate(
                    {
                        width:"0px",
                    },
                    {
                        duration: 1000 , queue: false
                    }

                    )
            }else{
                $(this).next().animate(
                {
                    width:"680px"
                },
                {
                    duration: 2000 , queue: false
                }
                );
            }

        });

        $("#runCode").click(function(){
            //content() 擷取網頁內容
            //find() 找到某特定內容
            // 雙重指定selector 用法 $("textarea#html-code") : 指在textarea裡面的 html-code (id)
           
            $("#result-frame").contents().find("body").html($("textarea#html-code").val());
            $("#result-frame").contents().find("head").html("<style>" + $("textarea#css-code").val() + "</style>");

            //contentWindow() 擷取網頁內容
            //eval()獲得方法
            document.getElementById('result-frame').contentWindow.eval($("textarea#js-code").val());
        });

    </script>

</body>
</html>
